docs/fabric_mod_json_spec.md

```diff
@@ -0,0 +1,130 @@
+- "value": The default language adapter uses this specific key as a string value of any of the following formats:
+  - "my.package.MyClass", which points to a class to be instantiated,
+  - "my.package.MyClass::thing", which points to a static field (contents returned) or method handle (for interface types, proxied automatically) named "thing".
+
+Fabric, by default, will run all "main" entrypoints (type ModInitializer), and all "client" (type ClientModInitializer) or "server" (type DedicatedServerModInitializer) entrypoints after that.
+
+#### NestedJarEntry
+
+An object with the following keys, of which only "file" is mandatory:
+
+- "file" - a string value pointing to a path from the root of the mod to a nested JAR which should be loaded alongside the outer mod JAR.
+
+#### Person
+
+Can be in one of two forms:
+
+- a string (assumed to resolve to a "name" key in the object),
+- an object.
+
+In the case of an object, the following keys are defined:
+
+- **name**: The real name, or username, of the person. Mandatory.
+- **contact** An optional ContactInformation object containing contact information pertaining to the person.
+
+#### VersionRange
+
+A string or array of string declaring supported version ranges. In the case of an array, an "OR" relationship is assumed - that is, only one range has to match for the collective range to be satisfied.
+
+In the case of all versions, `*` is a special string declaring that any version is matched by the range. In addition, exact string matches must be possible regardless of the version type.
+
+For semantic versions, the specification follows a rough subset of the [NPM semver](https://docs.npmjs.com/misc/semver) specification, in particular the following features are supported:
+
+- `=` as a leading character, or the lack of one, denoting an exact match,
+- Version ranges - a set of space-delimited comparators of the `>=`, `>`, `<=`, `<`, `=` or no prefix, following NPM behaviour and declaring an intersection of supported versions within the scope of the string,
+- X-Ranges,
+- Tilde Ranges,
+- Caret Ranges.
+
+### Mandatory fields
+
+- **id**: Contains the mod identifier - a string value matching the `^[a-z][a-z0-9-_]{1,63}$` pattern.
+- **version**: Contains the mod version - a string value, optionally matching the [Semantic Versioning 2.0.0](https://semver.org/) specification.
+
+### Optional fields (mod loading)
+
+- **environment**: For games with multiple environments - is a string value (or an array of string values) defining the environments the mod should be considered for loading on. Supported values are:
+  - "*" - all environments (default),
+  - "client" - the game client,
+  - "server" - the game dedicated server (integrated servers are not included here).
+- **entrypoints**: Contains an EntrypointContainer. If not present, assume empty object.
+- **jars**: Contains an array of NestedJarEntry objects. If not present, assume empty object.
```

Comment:
Link `EntrypointContainer` and `NestedJarEntry` to their respective sections in the markdown (no need to do if code blocks)